File: libs/typeorm/src/entities/ChannelModerationWarn.ts
import { Column, Entity, PrimaryColumn } from 'typeorm';

import { type SettingsType } from './ChannelModerationSetting';

@Entity('channels_moderation_warnings', { schema: 'public' })
export class ChannelModerationWarn {
  @PrimaryColumn('text', {
    primary: true,
    name: 'id',
    default: () => 'gen_random_uuid()',
  })
  id: string;

  @Column('text', { name: 'channelId' })
  channelId: string;

  @Column('text', { name: 'userId' })
  userId: string;

  @Column('text')
  reason: SettingsType;
}

File: libs/gomodels/channel_moderation_warn.go
package model

type ChannelModerationWarn struct {
	ID        string `gorm:"primary_key;column:id"      json:"id"`
	ChannelID string `gorm:"column:channelId;type:text" json:"channelId"`
	UserID    string `gorm:"column:userId;type:text"    json:"userId"`
	Reason    string `gorm:"column:reason;type:text"    json:"reason"`
}

func (ChannelModerationWarn) TableName() string {
	return "channels_moderation_warnings"
}
